pandas/core/indexing.py

```diff
@@ -1933,12 +1934,15 @@ def _ensure_iterable_column_indexer(self, column_indexer):
         """
         Ensure that our column indexer is something that can be iterated over.
         """
-        # Ensure we have something we can iterate over
         if is_integer(column_indexer):
             ilocs = [column_indexer]
         elif isinstance(column_indexer, slice):
             ri = Index(range(len(self.obj.columns)))
             ilocs = ri[column_indexer]
+        elif isinstance(column_indexer, np.ndarray) and is_bool_dtype(
```

Comment:
should this be is_list_like (and then `columns_indexer = np.asarray(column_indexer)` doesn't this *have* to be integer dtyped at this point? (if might be rejected before this is fine, but I think a plain list of ints can get here right)?

Reply:
yeah this should be correct. I thought more about performance here. We do not have to create an array in case of list or integer array, only in case of boolean

Reply:
its not about performance its about correctness. does a list-of-integers hit here?

Reply:
If they hit, they are already ilocs, so we can simply hit the else block